src/tools/Erase.ts

```diff
@@ -1,78 +1,70 @@
 
-export default class Erase implements EraseInterface {
-  // eslint-disable-next-line no-useless-constructor
+export default class Erase extends ToolClass {
   constructor (public readonly name: string,
-               public erase: string[],
                public readonly temporary: boolean) {
+    super()
   }
-  // eslint-disable-next-line
-  mouseDownAction = (event: CustomEvent, canvasElement: CanvasElement, layer: Konva.Layer, socket: WebSocket): void => {
-    canvasElement.data = [event.globalOffset.x, event.globalOffset.y]
-    canvasElement.id = uuid()
-    canvasElement.hasMoved = true
-    canvasElement.tracker = Tracker.REMOVAL
-    canvasElement.tool = {
+
+  mouseDownAction = (event: CustomEvent): void => {
+    this.enableTool()
+    this.resetCanvasEntity()
+    this.canvasEntity.modifyData = {
+      removals: []
+    }
+    this.canvasElement.tool = {
       name: this.name,
-      erase: this.erase,
       temporary: false
     }
-    this.findAndHide(event, canvasElement, layer)
-  }
-  // eslint-disable-next-line
-  mouseMoveAction = (event: CustomEvent, canvasElement: CanvasElement, layer: Konva.Layer, socket: WebSocket): void => {
-    this.findAndHide(event, canvasElement, layer)
+    this.findAndHide(event)
   }
 
-  mouseUpAction = (event: CustomEvent, canvasElement: CanvasElement, layer: Konva.Layer, socket: WebSocket): void => {
-    this.findAndHide(event, canvasElement, layer)
-    this.sendToWebSocket(canvasElement, socket)
+  mouseMoveAction = (event: CustomEvent): void => {
+    if (this.enabled) {
+      this.findAndHide(event)
+    }
   }
 
-  hideGroup = (layer: Konva.Layer, group?: string[]): void => {
-    if (group) {
-      group.forEach((groupId: string) => {
-        const group: Konva.Collection<Konva.Node> = layer.getChildren(node => node.attrs.id === groupId)
-        group.each(child => child.hide())
-        layer.batchDraw()
+  // eslint-disable-next-line
+  mouseUpAction = (event: CustomEvent): void => {
+    const data = this.canvasEntity.modifyData as RemovalData
+    if (this.enabled && data.removals) {
+      this.disableTool()
+      this.sendAndAddToState({
+        id: uuid(),
+        jti: this.canvasElement.jti,
+        modifyType: Tracker.REMOVAL,
+        modifyData: data,
+        canvasElements: [],
+        timestampModified: ISO.timestamp()
       })
     }
   }
 
-  findAndHide = (event: CustomEvent, canvasElement: CanvasElement, layer: Konva.Layer): void => {
-    const group = event.konvaPointerEvent.target.parent
-    if (group && group instanceof Konva.Group && group.attrs.id && canvasElement.tool.erase && !canvasElement.tool.erase.includes(group.attrs.id)) {
-      canvasElement.tool.erase.push(group.attrs.id)
-    }
-    this.hideGroup(layer, canvasElement.tool.erase)
+  hideGroup = (groupId: string): void => {
+    const foundGroup = this.layer.findOne((group: Konva.Group) => group.attrs.id === groupId)
+    if (foundGroup) { foundGroup.destroy() }
```

Comment:
When a shape is erased, we complete remove it from the layer. This will improve performance. If we want it back, we will just use the history data to re-render it